vagrant.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -euo pipefail
+_FINISH(){
+    RESULT=($?)
+    if [ ${RESULT} -eq 0 ]; then
+        echo ""
+        echo "Tests completed normally..."
+        vagrant destroy ${BOX}
+        echo ""
+    else
+        echo ""
+        echo "** Test failed.  Leaving '${BOX}' running for debug." 2>&1 | tee -a ${tee_file}
+        echo "** Be sure to halt or destroy prior to re-running the check" 2>&1 | tee -a ${tee_file}
+        echo "** Logs are stored at ${tee_file}"
+        echo ""
+        exit 1
+    fi
+}
+
+# When make calls bash, the real signals are not surfaced
+# correctly to trap.  So we trap on EXIT and then sort it
+# out in _FINISH
+trap _FINISH EXIT
+
+BOXES="fedora_atomic centos_atomic"
+
+is_running() {
+    status=$(vagrant status | grep ${BOX} | awk '{print $2}')
+    if [ ${status} == "running" ]; then
+        echo ""
+        echo "'${BOX}' is already running.  Perform a 'vagrant halt|destroy ${BOX}'"
```

Comment:
We can leave that to the user, no? I.e. expect them to know when the previous `make vagrant-check` is done before starting a new one. Or are you thinking of another use case here?